src/metabase/sync/sync_metadata/tables.clj

```diff
@@ -193,10 +193,10 @@
         (remove metabase-metadata/is-metabase-metadata-table?)
         (:tables db-metadata)))
 
-(mu/defn ^:private db->our-metadata :- [:set i/DatabaseMetadataTable]
+(mu/defn ^:private db->our-metadata :- [:set (ms/InstanceOf :model/Table)]
```

Comment:
the `i/DatabaseMetadataTable` schema is now a closed one, so it was failing because the table from our metadata includes an `:id`. rather than adding an `:id` to `i/DatabaseMetadataTable`, I think it makes more sense that our-metadata has its own schema here. This is how we do in sync fields as well.